src/modules/navigator/mission_block.cpp

```diff
@@ -359,6 +359,12 @@ MissionBlock::is_mission_item_reached_or_completed()
 
 			}
 
+			// prevent lateral guidance from loitering at a waypoint as part of a mission landing if the altitude
+			// is not achieved.
```

Comment:
I was thinking of how to avoid, needing to replicate the logic here for the mission item, since it is already set in the position setpoint triplet. Here is mi radical thought. In navigator there exists the get_altitude_acceptance_radius. In the navigator you have also access to the position setpoint. So there you could actually ready out the altitude radius in the current position setpoint, and if it is not NaN, just pass it along, else use the default one as calculated in the function as well. Then here you only need to cal the navigator->get_altitude_acceptance_radius function and you do not need to duplicate the logic. Hope this works, not sure yet.

Reply:
@sfuhrer Agree, and currently trying out the proposed solution.